libraries/AP_GPS/AP_GPS_NOVA.cpp

```diff
@@ -44,36 +44,75 @@ AP_GPS_NOVA::AP_GPS_NOVA(AP_GPS &_gps, AP_GPS::GPS_State &_state,
 {
     nova_msg.nova_state = nova_msg_parser::PREAMBLE1;
 
-    const char *init_str = _initialisation_blob[0];
-    const char *init_str1 = _initialisation_blob[1];
-    
-    port->write((const uint8_t*)init_str, strlen(init_str));
-    port->write((const uint8_t*)init_str1, strlen(init_str1));
+    if(get_type() == AP_GPS::GPS_TYPE_NOVA || get_type() == AP_GPS::GPS_TYPE_NOVA_DUAL) {
+        const char *init_str = _nova_initialisation_blob[0];
+        const char *init_str1 = _nova_initialisation_blob[1];
+
+        port->write((const uint8_t*)init_str, strlen(init_str));
+        port->write((const uint8_t*)init_str1, strlen(init_str1));
+    } else {
+        const char *init_str = _unicore_initialisation_blob[0];
+        const char *init_str1 = _unicore_initialisation_blob[1];
+
+        port->write((const uint8_t*)init_str, strlen(init_str));
+        port->write((const uint8_t*)init_str1, strlen(init_str1));
+    }
 }
 
-const char* const AP_GPS_NOVA::_initialisation_blob[6] {
+const char* const AP_GPS_NOVA::_nova_initialisation_blob[7] {
     "\r\n\r\nunlogall\r\n", // cleanup enviroment
     "log bestposb ontime 0.2 0 nohold\r\n", // get bestpos
     "log bestvelb ontime 0.2 0 nohold\r\n", // get bestvel
     "log psrdopb onchanged\r\n", // tersus
     "log psrdopb ontime 0.2\r\n", // comnav
-    "log psrdopb\r\n" // poll message, as dop only changes when a sat is dropped/added to the visible list
+    "log psrdopb\r\n", // poll message, as dop only changes when a sat is dropped/added to the visible list
+    "log dualantennaheadinga ontime 0.2\r\n"
 };
 
+const char* const AP_GPS_NOVA::_unicore_initialisation_blob[7] {
+    "\r\n\r\nunlogall\r\n", // cleanup enviroment
+    "log bestposb ontime 0.2\r\n", // get bestpos
+    "log bestvelb ontime 0.2\r\n", // get bestvel
+    "log psrdopb onchanged\r\n", // tersus
+    "log psrdopb ontime 0.2\r\n", // comnav
+    "log psrdopb\r\n", // poll message, as dop only changes when a sat is dropped/added to the visible list
+    "log headingb ontime 0.2\r\n"
+};
 // Process all bytes available from the stream
 //
 bool
 AP_GPS_NOVA::read(void)
 {
     uint32_t now = AP_HAL::millis();
-
-    if (_init_blob_index < (sizeof(_initialisation_blob) / sizeof(_initialisation_blob[0]))) {
-        const char *init_str = _initialisation_blob[_init_blob_index];
-
-        if (now > _init_blob_time) {
-            port->write((const uint8_t*)init_str, strlen(init_str));
-            _init_blob_time = now + 200;
-            _init_blob_index++;
+    if(get_type() == AP_GPS::GPS_TYPE_NOVA) {
```

Comment:
this could be made considerably simpler by using the get_type() to setup local variables for the array of init blobs and the array length, then having just one copy of the blob send code. It is good to minimise code duplication.